man/udevadm.xml

```diff
@@ -382,6 +382,21 @@
             then each matching result is ORed, that is, all children of each specified device are triggered.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><option>--database-nomatch</option></term>
+          <listitem>
+            <para>Trigger devices without a corresponding udev database entry only. This can potentially save a
+            significant amount of time compared to re-triggering all devices in the system and e.g. can be used to
+            optimize boot time.</para>
```

Comment:
> so if you add a discussion of the pros here, you need to mention the cons too, otherwise this just will make people wonder why this is not the default. i.e. it needs to say that using this means is only safe if during the entire preceeding lifetime of udev and its database the same udev ruleset and code has been used, i.e. that it only works in initrd-less scenarios.

I agree, I should also mention the limitations of this option here.

> (which of course then raises the question: maybe we should have `--database-nomatch-auto` or so as a switch that is like `--database-match` iff we are run in an initrd or in an initrd-less system [which I am not sure we have a nice way to detect] and otherwise suppress it). Because that would be a safe optimization then. If you follow my thoughts on this)

I agree, being able to detect that would be nice, since this could really become the default in systemd-udev-trigger.service then.

However, I don't know if you still want to consider this PR, since https://github.com/systemd/systemd/pull/22662 replaces this PR implementing the same thing slightly different. So maybe we want to consider this in the scope of https://github.com/systemd/systemd/pull/22662. Otherwise we could proceed documenting the limitations of this option and add something like `--database-nomatch-auto` with a subsequent PR.

CC: @yuwata